google-cloud-storage/src/test/java/com/google/cloud/storage/TimestampCodecPropertyTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage;
+
+import static com.google.cloud.storage.JqwikTest.report;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.storage.Conversions.Codec;
+import com.google.cloud.storage.jqwik.StorageArbitraries;
+import com.google.protobuf.Timestamp;
+import java.time.OffsetDateTime;
+import net.jqwik.api.Arbitrary;
+import net.jqwik.api.ArbitrarySupplier;
+import net.jqwik.api.Example;
+import net.jqwik.api.ForAll;
+import net.jqwik.api.Property;
+import net.jqwik.api.providers.TypeUsage;
+
+final class TimestampCodecPropertyTest {
+
+  @Example
+  void edgeCases() {
+    report(TypeUsage.of(Timestamp.class), StorageArbitraries.timestamp());
+  }
+
+  @Property(tries = 50_000)
+  void timestampCodecShouldRoundTrip(@ForAll(supplier = Supp.class) Timestamp ts) {
+    Codec<OffsetDateTime, Timestamp> codec = GrpcConversions.INSTANCE.timestampCodec;
+    OffsetDateTime decode = codec.decode(ts);
+    Timestamp encode = codec.encode(decode);
+
+    assertThat(encode).isEqualTo(ts);
+  }
+
+  private static final class Supp implements ArbitrarySupplier<Timestamp> {
+    @Override
+    public Arbitrary<Timestamp> get() {
+      return StorageArbitraries.timestamp();
```

Comment:
Why did you need this helper static method?

Reply:
Rather than the `Arbitrary<Timestamp>` from `StorageArbitraries` being registered globally, this method provides the resolution handle to jqwik so it knows how to generate arbitraries for the test method.

Above on line 41 `@ForAll(supplier = Supp.class)` points to this class as the means to resolve the Arbitrary<Timestamp>.

Reply:
If we'd rather it be registered globally, I can do that too.

Reply:
oh, this is just to test the timestampCodec specifically; wouldn't you need a timestamp supplier for other tests such as bucket/ object metadata? I think that's what is confusing me. This is more of a question that can be answered after PR approval. Thanks Ben

Reply:
Other things would also need the timestamp supplier, which so far has been accessed and composed directly via `StorageArbitraries#timestamp()`.

This supplier class is effectively wiring so that jqwik finds that method for the test.